### PR TITLE
Jan/support children matching

### PR DESCRIPTION
**Pull request by janmyszkier**

### CODE REVIEW

**Comment by youngskilled** [comment]
---

Reply ABOVE THIS LINE to add a comment to this case​

YS | Sep 23, 2016 08:27AM UTC:​
Thank you for submitting your request. We have received your request and are working on responding to you as soon as possible. If you have any additional information to add to this case, please reply to this email.

Thanks in advance for your patience and support.

For your reference this is Case #: 11461

[[4a866a767d11ba629b2c08b0144f434a78072a59-799129972]]

File: dev/js/ysFilter.js [hunk 1]
```diff
@@ -87,12 +97,24 @@
 				paramTypes = $this.filter.settings.filter;
 				totalItems = $this.filter.filter;
 
+				var categoryRegex = '^' + $this.set.category;
```
Comment (richardnadj) [inline dev/js/ysFilter.js]: This should be the default and only written over further down. This should however be for exact matching of the categories.

File: dev/js/ysFilter.js [hunk 2]
```diff
@@ -87,12 +97,24 @@
...
 				for(var category in categories) {
-					var testCorrectCategory = new RegExp('^' + category + '$');
-					if(testCorrectCategory.test($this.set.category)) {
+
+					if($this.set.alsoMatchChildrenCategories) {
+						/* Match all children categories */
+						categoryRegex = '^' + $this.set.category;
+					} else {
+						/* Only match current category */
```
Comment (richardnadj) [inline dev/js/ysFilter.js]: Remove these lines in favour of using the default

**Review by richardnadj — COMMENTED** [review]
Good other than the comment

**Review by richardnadj — APPROVED** [review]

**Pull request merged**